src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -3821,25 +3827,43 @@ let parties_witnesses ~constraint_constants ~state_body ~fee_excess
         let mk_next_commitment (parties : Parties.t) =
           empty_if_last (fun () -> Parties.commitment parties)
         in
+        let mk_next_full_commitment (parties : Parties.t) =
```

Comment:
Nit: combine this with the above function and return a tuple. They're never called in isolation, and this setup will recompute the next commitment.